### PR TITLE
Add XGBoostClassifier

### PR DESCRIPTION
**Pull request by aemr3**
## Summary

Added XGBoostClassifier class and a test case

### CODE REVIEW

**Comment by xmatthias** [comment]
sorry - accidentally closed this PR as i merged the continual-learning branch.

I've now reopened it and changed the destination to "develop" - so we can accept it directly

File: freqtrade/freqai/prediction_models/XGBoostClassifier.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+        :do_predict: np.array of 1s and 0s to indicate places where freqai needed to remove
+        data (NaNs) or felt uncertain about data (PCA and DI index)
+        """
+
+        (pred_df, dk.do_predict) = super().predict(unfiltered_df, dk, **kwargs)
+
+        le = LabelEncoder()
+        label = dk.label_list[0]
+        labels_before = list(dk.data['labels_std'].keys())
+        labels_after = le.fit_transform(labels_before).tolist()
```
Comment (robcaulk) [inline freqtrade/freqai/prediction_models/XGBoostClassifier.py:80]: Can we incorporate the `predict_proba` function as well? You can see how we do it in `BaseClassifierModel.py` and emulate it here. Users will be expecting the probabilities to come back in addition to the classification - so we should try to stick with that paradigm. 

It seems `XGBClassifier` has the `predict_proba` method similar to the others: 
https://xgboost.readthedocs.io/en/stable/python/python_api.html?highlight=xgbclassifier#xgboost.XGBClassifier.predict_proba

Comment (aemr3) [reply freqtrade/freqai/prediction_models/XGBoostClassifier.py:80]: Actually, predict_proba is already emulated in base class and appended in this pred_df dataframe. Column names are also transformed. Here is an example output from debugger:

```
(Pdb) pred_df
    &s-trend      down        up
0         up  0.050633  0.949367
1         up  0.049386  0.950614
2         up  0.069725  0.930275
3         up  0.069076  0.930924
4         up  0.015851  0.984149
..       ...       ...       ...
994       up  0.001230  0.998770
995       up  0.033075  0.966925
996     down  0.685019  0.314981
997     down  0.652857  0.347143
998       up  0.050721  0.949279

[999 rows x 3 columns]
```

**Review by robcaulk — APPROVED** [review]
Thanks for this!

**Comment by Bilal1422** [comment]
Problem solved sirSent from my Samsung Galaxy smartphone.
-------- Original message --------From: Robert Caulk ***@***.***> Date: 13/09/2022  2:08 am  (GMT+03:30) To: freqtrade/freqtrade ***@***.***> Cc: Subscribed ***@***.***> Subject: Re: [freqtrade/freqtrade] Add XGBoostClassifier (PR #7390) 
Merged #7390 into develop.

—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you are subscribed to this thread.Message ID: ***@***.***>
[
{
***@***.***": "http://schema.org",
***@***.***": "EmailMessage",
"potentialAction": {
***@***.***": "ViewAction",
"target": "https://github.com/freqtrade/freqtrade/pull/7390#event-7371278689",
"url": "https://github.com/freqtrade/freqtrade/pull/7390#event-7371278689",
"name": "View Pull Request"
},
"description": "View this Pull Request on GitHub",
"publisher": {
***@***.***": "Organization",
"name": "GitHub",
"url": "https://github.com"
}
}
]

**Pull request merged**